Dockerfile

```diff
@@ -1,50 +1,57 @@
-FROM ubuntu:18.04
+# === BUILDER IMAGE ===
+FROM alpine:3.12 as builder
 USER root
-
 WORKDIR /asn1_codec
-
 VOLUME ["/asn1_codec_share"]
 
-# Add build tools.
-RUN apt-get update && apt-get install -y software-properties-common wget git make gcc-7 g++-7 gcc-7-base && update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-7 100 && update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-7 100
-
-# Install cmake.
-RUN apt-get install -y cmake
+# update the package manager
+RUN apk update
 
-# install libtool and automake
-RUN apt-get install -y automake libtool
-
-# Install librdkafka.
-RUN apt-get install -y sudo
-RUN wget -qO - https://packages.confluent.io/deb/7.3/archive.key | sudo apt-key add -
-RUN add-apt-repository "deb [arch=amd64] https://packages.confluent.io/deb/7.3 stable main"
-RUN add-apt-repository "deb https://packages.confluent.io/clients/deb $(lsb_release -cs) main"
-RUN apt update
-RUN apt-get install -y libsasl2-modules libsasl2-modules-gssapi-mit libsasl2-dev libssl-dev 
-RUN apt install -y librdkafka-dev
+# add build dependencies
+RUN apk add --upgrade --no-cache --virtual .build-deps \
+    cmake \
+    g++ \
+    make \
+    bash \
+    automake \
+    libtool \
+    autoconf \
+    librdkafka \
+    librdkafka-dev \
+    flex \
+    bison
 
 # Install pugixml
 ADD ./pugixml /asn1_codec/pugixml
 RUN cd /asn1_codec/pugixml && mkdir -p build && cd build && cmake .. && make && make install
 
 # Build and install asn1c submodule
 ADD ./asn1c /asn1_codec/asn1c
-RUN cd /asn1_codec/asn1c && aclocal && test -f configure || autoreconf -iv && ./configure && make && make install
+RUN cd asn1c && test -f configure || autoreconf -iv && ./configure && make && make install
```

Comment:
Can we make this consistent with the pugxml install by installing it inside image directory /asn1_codec/asn1c?

Reply:
This is already getting installed into the /asn1_codec/asn1c directory.

The `WORKDIR /asn1_codec` instruction was executed prior to this. We are using a local reference to cd into the /asn1c_codec/asn1c directory on line 30. Should we change this to an absolute reference?